prov/rxm/src/rxm_domain.c

```diff
@@ -447,8 +447,13 @@ static int rxm_mr_close(fid_t fid)
 	if (rxm_mr->domain->util_domain.info_domain_caps & FI_ATOMIC)
 		rxm_mr_remove_map_entry(rxm_mr);
 
-	ret = fi_close(&rxm_mr->msg_mr->fid);
-	if (ret)
+    // release the gdrcopy if present
+    if (rxm_mr->hmem_handle) {
```

Comment:
Being pedantic. This branch should also check iface == CUDA

Also alignment is off.